ymer/src/main/java/com/avanza/ymer/MongoDocumentCollection.java

```diff
@@ -118,12 +140,25 @@ public void update(Document newVersion) {
 	}
 
 	@Override
-	public void updateById(Object id, Map<String, Object> fieldsAndValuesToSet) {
-		fieldsAndValuesToSet.entrySet().stream().map(entry -> Updates.set(entry.getKey(), entry.getValue())).reduce(Updates::combine)
-				.ifPresent(updates -> {
-					UpdateResult updateResult = collection.updateOne(Filters.eq(id), updates);
-					idValidator.validateUpdatedExistingDocument("update", updateResult, new Document("_id", id));
-				});
+	public void bulkWrite(Consumer<BulkWriter> bulkWriter) {
+		List<WriteModel<Document>> writeModels = new ArrayList<>();
+		bulkWriter.accept((ids, fieldsToSet) -> {
+			Bson updates = toUpdates(fieldsToSet);
+			if (ids.isEmpty()) {
+				log.warn("Skipping updates because no ids provided");
+			} else if (updates == null) {
+				log.warn("Skipping updates because no fields to update provided");
+			} else {
+				Bson filter = Filters.in("_id", ids);
+				UpdateManyModel<Document> updateManyModel = new UpdateManyModel<>(filter, updates);
+				writeModels.add(updateManyModel);
+			}
+		});
+		if (writeModels.isEmpty()) {
+			log.warn("Skipping bulkWrite because no updates provided");
```

Comment:
Maybe lower the logging level of this statement? Now with the refactoring of using a `Consumer<BulkWriter>`, this gets called even if there are no updates to be done, for example for a collection whose `_instanceId` are already correct, leading to some unnecessary warning-logging.